src/main/java/com/arcbees/bourseje/client/admin/dashboard/AdminDashboardView.java

```diff
@@ -114,18 +94,16 @@ public void f() {
     }
 
     @Override
-    public void setNumberOfVotesForCandidate(CandidateResult candidateResult) {
-        SpanElement element = numberOfVoteElements.get(candidateResult.getCandidateName());
-
-        if (element == null) {
-            GQuery.console.error("Candidate not found: " + candidateResult.getCandidateName());
-        } else {
-            element.setInnerText(String.valueOf(candidateResult.getNumberOfVotes()));
-        }
+    public void setCurrentState(VoteState currentState) {
+        this.currentState.setInnerText(currentState.toString());
     }
 
     @Override
-    public void setCurrentState(VoteState currentState) {
-        this.currentState.setInnerText(currentState.toString());
+    public void setCandidates(List<Candidate> candidatesList, Map<String, Integer> candidateResults) {
+        for(Candidate candidate : candidatesList) {
+            CandidateWidget candidateWidget = new CandidateWidget(candidate, candidateResults.get(candidate.getName()));
+
+            $(candidates).before(candidateWidget.asWidget().getElement());
```

Comment:
I think `candidates` is confusing. I would expect the `CandidateWidget`s to be inside the HTMLPanel